pkg/kubectl/cmd/apply_test.go

```diff
@@ -213,6 +216,59 @@ func TestApplyObject(t *testing.T) {
 	}
 }
 
+func TestApplyRetry(t *testing.T) {
+	initTestErrorHandler(t)
+	nameRC, currentRC := readAndAnnotateReplicationController(t, filenameRC)
+	pathRC := "/namespaces/test/replicationcontrollers/" + nameRC
+
+	firstPatch := true
+	retry := false
+	f, tf, codec := NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.Client = &fake.RESTClient{
+		Codec: codec,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == pathRC && m == "GET":
+				bodyRC := ioutil.NopCloser(bytes.NewReader(currentRC))
```

Comment:
nit:  Would you also check that the cmd calls this twice and applies the patch when retrying the read.  e.g. return a slightly different currentRC the second time this is called, and verify in the PATCH that the second read was used for the patch.

Let me know if this is excessively complicated to do.